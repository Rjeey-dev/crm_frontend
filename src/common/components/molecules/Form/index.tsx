import React, {ReactNode, ReactNodeArray} from "react";

interface IProps {
    children: ReactNodeArray | ReactNode,
    action?: string,
    classes?: string,
    onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void,
}

function Form(props: IProps) {
    return <form onSubmit={props.onSubmit} action={props.action} className={props.classes}>
        {props.children}
    </form>
}

Form.defaultProps = {
    action: '#'
};

export default Form;
