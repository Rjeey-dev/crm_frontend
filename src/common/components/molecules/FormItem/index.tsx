import React from 'react';

import Field from "common/components/atoms/Field/index";

interface IProps {
    meta: any,
    input: any,
}

const FormItem = ({ meta, input, ...props }: IProps) => {
    const fieldProps = {...props, initial: meta.initial, ...input, invalid: meta.touched && !!meta.error, error: meta.error,};

    if (fieldProps.type === 'text' || fieldProps.type === 'textarea') {
        return <Field {...fieldProps} >{fieldProps.children}</Field>
    }

    return <Field {...fieldProps} >{fieldProps.children}</Field>
};

export default FormItem;
