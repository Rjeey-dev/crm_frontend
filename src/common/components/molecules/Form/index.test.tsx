import {shallow} from "enzyme";
import React from 'react';

import FormItem from '../FormItem/index';
import Form from "./index";

it("Renders <Form/> with correct properties", () => {
    const output = shallow(
        <Form action='login' classes='class' onSubmit={jest.fn()} name='loginForm'>
            <FormItem meta={{}} input={{}}/>
        </Form>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <Form/> with only required props", () => {
    const output = shallow(
        <Form name='loginForm' onSubmit={jest.fn()}>
            <FormItem meta={{}} input={{}}/>
            <FormItem meta={{}} input={{}}/>
        </Form>
    ).html();

    expect(output).toMatchSnapshot();
});