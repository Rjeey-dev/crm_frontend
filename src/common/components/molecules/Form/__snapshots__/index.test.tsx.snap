// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Form/> with correct properties 1`] = `"<form action=\\"login\\" class=\\"class\\" name=\\"loginForm\\"><div class=\\"form-group\\"><div class=\\"input-group\\"><input aria-describedby=\\"undefinedError\\" class=\\"form-control\\"/><div class=\\"input-group-append\\"><span class=\\"input-group-text\\"><i class=\\"mdi mdi-check-circle\\"></i></span></div></div></div></form>"`;

exports[`Renders <Form/> with only required props 1`] = `"<form action=\\"#\\" name=\\"loginForm\\"><div class=\\"form-group\\"><div class=\\"input-group\\"><input aria-describedby=\\"undefinedError\\" class=\\"form-control\\"/><div class=\\"input-group-append\\"><span class=\\"input-group-text\\"><i class=\\"mdi mdi-check-circle\\"></i></span></div></div></div><div class=\\"form-group\\"><div class=\\"input-group\\"><input aria-describedby=\\"undefinedError\\" class=\\"form-control\\"/><div class=\\"input-group-append\\"><span class=\\"input-group-text\\"><i class=\\"mdi mdi-check-circle\\"></i></span></div></div></div></form>"`;
